### ISSUE TITLE
Doesn't support -e git+https:// type links

### ISSUE THREAD

**Issue by jjangsangy** [kennethreitz/pipenv#414]
Hey @kennethreitz, was listening to you on Talk Python To Me so I played around with pipenv. I'm filing a couple of issues including this one.

Even when running `pipenv lock --requirements` there doesn't seem to be a way for these types of links to be generated from the `requirements.txt` file

**Comment by nateprewitt** [comment]
Hey @jjangsangy, could you clarify in a little more detail what you're expecting vs. what you're seeing?

As an example, I can `pipenv install git+https://github.com/requests/requests.git#egg=requests` and it will create a Pipfile like this.

```
[packages.requests]
git = "https://github.com/requests/requests.git"
editable = true
```

When I run `pipenv lock --requirements` I get:
```
-e git+https://github.com/requests/requests.git#egg=requests
```

**Comment by jjangsangy** [comment]
So in my requirements file I have this which pulls from a specific branch.

`-e git+https://github.com/jjangsangy/python-goose.git@develop#egg=goose-extractor`

**Comment by jjangsangy** [comment]
What ends up in my Pipfile is
`goose-extractor = "*"`

**Comment by jjangsangy** [comment]
I think the error actually comes from `--editable`

**Comment by erinxocon** [comment]
I can confirm this issue.  When you are using `pipenv lock --requirements` everything is exported as expected, however when you create an environment fresh that only has a requirements file, `pipenv` will convert the requirements file into a Pipfile, but it doesn't translate custom VCS links properly.  It just takes the egg name and puts '*' as the version instead of creating a subsection.  I ran into the same problem.  I'm working on a patch!

**Comment by nateprewitt** [comment]
Thanks @erinxocon! I did some digging yesterday and you're going to need to extract the information for editable installs from the `InstallRequirement` returned from pip. The default `__str__` method doesn't return an equivalent string to the requirements.txt entry. This shouldn't be too difficult to fix, but will require some analysis on each dependency.

**Comment by erinxocon** [comment]
I believe #416 should fix this!  I haven't tried any other VCS's except git over https, but it does convert my requirements file correctly that contain links to github repos!  

It doesn't currently handle the editable prefix in that when it creates the packages.repo section from a VCS link in a requirements file, editable=True is not present when -e is.  I'll have to figure that out if we feel like it's necessary.

**Comment by erinxocon** [comment]
and now it can handle editable links!

**Comment by nateprewitt** [comment]
Alright, now that #416 is merged, I think this should be fixed in the next release. @jjangsangy feel free to test out the master branch if you'd like to verify things work before hand.

I'll try to get a release out later this week.

**Comment by nateprewitt** [comment]
Just FYI, I've finally released 5.1.2, so this should be fixed going forward. Thanks again for your work @erinxocon and for everyone's patience!
